src/engine/BMButton.php

```diff
@@ -128,7 +129,10 @@ public function load($recipe, $name = NULL, $isRecipeAltered = FALSE) {
             }
         }
 
-        $this->validate_recipe($recipe);
+        if (('CustomBM' === $name) || $isRecipeAltered) {
+            $this->validate_recipe($recipe);
+        }
+
```

Comment:
Why is this change necessary?

Reply:
I noted that the recipe validation step was happening *every* time the button list was being loaded, which is both cumbersome and unnecessary.

In view of your comments below regarding late addition of sides via auxiliary, I'll need to find a different way of restricting validation to CustomBM recipes only at the time the recipe is specified.

Reply:
Hmm.  Please don't make logic changes in this PR that aren't needed to fix this PR.

Reply:
I'd have to disagree here. We're adding more validation logic here, some of which is quite computationally intensive. I think it's valid to consider whether we need to run the new logic against every button recipe every time that it is loaded (i.e., on the game creation page).

Reply:
Note that *every* button recipe is being validated twice each time that we load the game creation page, in case you were not aware of this.

Reply:
But this change makes it so that every button recipe is validated zero times, ever.  We can add new button recipes that violate constraints, and we will never find out about it until something breaks in some weird way.

I feel like we have to think pretty carefully about the safety of not validating recipes that come from the database.  I'd rather not think about that safety while also thinking about this PR.

Do you have evidence that the change in complexity makes the site unusable?  Or is it just bothering you philosophically?